src/mscorlib/src/System/Array.cs

```diff
@@ -1073,6 +1073,42 @@ public static T[] Empty<T>()
             return Array.FindIndex(array, match) != -1;
         }
 
+        public static void Fill<T>(T[] array, T value)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = value;
+            }
+        }
+
+        public static void Fill<T>(T[] array, int startIndex, int count, T value)
```

Comment:
@terrajobst, this API shape is a little concerning to me.  Suppose T == int.  Then going from the shorter overload to the longer overload changes the interpretation of the int as the second argument from value to startIndex.  Did we consider reordering the args to:
```C#
public static void Fill<T>(T[] array, T value);
public static void Fill<T>(T[] array, T value, int startIndex, int count);
```
?  I do understand that it's common to have array/offset/count all next to each other as args; maybe that consistency is more important?